### PR TITLE
pam_mkhomedir: use UMASK from /etc/login.defs

### PR DESCRIPTION
**Pull request by dshein-alt**
Follow the example of useradd(8) and set the file mode creation mask to
the value of UMASK configuration item from /etc/login.defs when umask=
option is not specified.

### CODE REVIEW

**Review by ldv-alt — APPROVED** [review]

**Comment by t8m** [comment]
Please note that recently the shadow utilities added a new option HOME_MODE to login.defs which overrides the UMASK for the home directory creation. I think it should be implemented here as well.

**Comment by t8m** [comment]
There is one caveat - the shadow utilities do not apply the HOME_MODE and the UMASK to the files copied from /etc/skel. It is questionable whether the pam_mkhomedir should continue doing it or not. The HOME_MODE should definitely be applied only to the mode of the home directory itself.

**Comment by ldv-alt** [comment]
> Please note that recently the shadow utilities added a new option HOME_MODE to login.defs which overrides the UMASK for the home directory creation. I think it should be implemented here as well.

I think it would make sense to add HOME_MODE support but it should rather be done in a separate follow-up patch.

**Comment by dshein-alt** [comment]
I've updated code to follow shadow/useradd behavior in case of umask option was not specified for pam_umask, so permissions for created user home directory is set in accordance to value of HOME_MODE and preempts UMASK if both are defined if /etc/login.defs. At the same time permissons for files and folders that copied fron /etc/skel is masked with default umask value.

Now home directory for users that loged in to system using pam creates with same permissions as for local one.

**Comment by t8m** [comment]
We do not allow merge commits. Could you please just cleanup them and rebase your changes on top of the current master?

**Comment by dshein-alt** [comment]
Got it.
I'll clean it up

File: modules/pam_mkhomedir/pam_mkhomedir.8.xml [hunk 1]
```diff
@@ -91,8 +91,14 @@
         <listitem>
           <para>
             The file mode creation mask is set to
-            <replaceable>mask</replaceable>. The default value of mask is
-            0022.
+            <replaceable>mask</replaceable>. The default value of mask
```
Comment (t8m) [inline modules/pam_mkhomedir/pam_mkhomedir.8.xml:94]: Please use something like:
```
           The default value of mask
            is 0022. If this option is not specified, then the permissions
            of the created user home directory is set to the value of
            <option>HOME_MODE</option> configuration item from
            <filename>/etc/login.defs</filename>. If there is no such
            configuration item then the value is computed from the
            value of <option>UMASK</option> in the same file. If
            there is no such configuration option either the default
            value of 0755 is used for the mode.
```

**Review by t8m — CHANGES_REQUESTED** [review]
I am sorry for neglecting your PR for so long time. Can you please adjust the documentation? Otherwise it looks good to me.

File: modules/pam_mkhomedir/pam_mkhomedir.8.xml (hunk 1 above)
Comment (dshein-alt) [reply modules/pam_mkhomedir/pam_mkhomedir.8.xml:94]: Updated documentation for pam_mkhomedir and mkhomedir_helper

File: modules/pam_mkhomedir/mkhomedir_helper.8.xml [hunk 2]
```diff
@@ -43,7 +46,9 @@
     <para>
       The default value of <replaceable>umask</replaceable> is 0022 and the
       default value of <replaceable>path-to-skel</replaceable> is
-      <emphasis>/etc/skel</emphasis>.
+      <emphasis>/etc/skel</emphasis>. The default value of
+      <replaceable>home_mode</replaceable> is computed form the value of
```
Comment (t8m) [inline modules/pam_mkhomedir/mkhomedir_helper.8.xml:50]: Typo: from

**Review by t8m — CHANGES_REQUESTED** [review]
One last nit

File: modules/pam_mkhomedir/mkhomedir_helper.8.xml (hunk 2 above)
Comment (dshein-alt) [reply modules/pam_mkhomedir/mkhomedir_helper.8.xml]: Fixed.
Thank you!

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged. Thank you for the contribution!

**Comment by ldv-alt** [comment]
Sorry, I missed the second iteration completely,
I really don't like the way the author name and email were mangled as result of squashing.
If it was github that failed to do a proper squashing, we should use this github feature with care.

**Comment by dshein-alt** [comment]
I suppose that name and email  wasn't fetched during squashing due to my Github profile settings.
I've probably missed some steps during registration.

**Pull request merged**